meetings/http/2017-03-14/minutes.md

```diff
@@ -1,24 +1,65 @@
 # HTTP stream meeting 2
 
-### Time: Wednesday March 15th 
-* 6pm CET 
-* 5pm UK 
-* 1pm EST 
-* 10am PST
-            
-            
-### WebEx and Agenda/Minutes doc:
-            
-WebEx:      URL:        https://ibm2.webex.com/ibm2-en/j.php?MTID=m59d2772a43f192bf64ae3bdbd5c77faf
-            password:   swift-http
-        
-Agenda:     https://docs.google.com/document/d/1GTm7mJH2vCAK-Ts4M3t8C3boJ18XqiFnCRuyp55eKJE/edit?usp=sharing
-
-If you'd like to add anything to the agenda ahead of time, please update the agenda document in Google Docs.
 
 ### Initial Agenda:
-* Review of draft HTTP API Proposal 
+* Review of draft HTTP API Pitch 
+  * https://docs.google.com/document/d/1ErHA8RXLiTGvOPHNY0KtbWCvwR1IUz_2ffi7CxCUo_4/edit
 * Discussion of key points:
   * Use of URLRequest/Response
   * Value Types vs. Reference Types
 
+### Attendees:
+* Chris Bailey
+* Ben Cohen
+* Carl Brown
+* Stevey Brown
+* Tony Parker
+* Gregor Milos
+* Shmuel Kallner
+* Alex Blewitt
+* Allen
+* Chuck Goddard
+* George
+* Johannes Weiss
+* Ludovic Dewailly
+* Marcin Kliks
+* Nasser Ebrahim
+* Steve Algernon
+* Michael Papp
+* Thomas Catterall
+
+### Minutes:
+
+_Use of value types vs. reference types for request and response_ 
+* General consensus is in favour of using a value types for request and response.
+  * The argument that the should be references as they are passed around and mutated really don’t hold
+  * It should be noted that the reason why there is still NSURLRequest/Response (reference types) was because a subclass exists, rather than the intend that they should be reference types.
+  * The use of value types will help with reference cycles in closures (like middlewares) which you otherwise need to use weak self for.
+  * Should also help with reducing some of the ARC overhead
```

Comment:
> The discussion about ARC seems like a moot point since requests and responses are not passed between threads in a dispatch based web server

It is not a moot point, I'm a little confused. You must be aware that the problem with ARC is that the sync overhead occurs whether or not threads are actually in use? The compiler simply doesn't know about that and hence has to apply the overhead *always*. Which is unlike struct copying in which the compiler actually has the ability to optimize a lot of copying away (there is no real reason to ever copy a let struct the way down the call hierarchy, etc). It is just a matter of fact.

Yes, the optimizer can try to reduce the actual ARC operations (presumably it does way to many of them now, partially to please ObjC), but the fact stands that passing an ARC reference is a synchronized operation and that a synchronized operation in todays platforms can be significantly more expensive than copying 10 or even 100 times the data.

If you *know* that a request is not passed between threads, an `UnsafePointer<struct>` may be the way to go. This just passes the reference and doesn't incur the ARC overhead (which again is not incrementing a counter, but *atomically* incrementing a counter). Or maybe Swift 4 gets a way to mark references as 'single-threaded', that would be an option too?

I agree with you that performance is not everything and that API usability is important too, but since we are talking about low level frameworks here, not about the thing the actual developer is going to see (as you are going to have nice-Kitura APIs on top for them), maybe it is worth doing a proper evaluation.
Or we are talking about a higher-level framework in the sense of the Python `http` module which probably means that most performance sensitive frameworks will avoid it, as the `http` module.